modules/markup/markdown/goldmark.go

```diff
@@ -197,18 +196,62 @@ func (g *ASTTransformer) Transform(node *ast.Document, reader text.Reader, pc pa
 			if css.ColorHandler(strings.ToLower(string(colorContent))) {
 				v.AppendChild(v, NewColorPreview(colorContent))
 			}
-		case *ast.Emphasis:
-			// check if inside blockquote for attention, expected hierarchy is
-			// Emphasis < Paragraph < Blockquote
-			blockquote, isInBlockquote := n.Parent().Parent().(*ast.Blockquote)
-			if isInBlockquote && !attentionMarkedBlockquotes.Contains(blockquote) {
-				fullText := string(n.Text(reader.Source()))
-				if fullText == AttentionNote || fullText == AttentionWarning {
-					v.SetAttributeString("class", []byte("attention-"+strings.ToLower(fullText)))
-					v.Parent().InsertBefore(v.Parent(), v, NewAttention(fullText))
-					attentionMarkedBlockquotes.Add(blockquote)
-				}
+		case *ast.Blockquote:
+			// We only want attention blockquotes when the AST looks like:
+			// Text: "["
+			// Text: "!TYPE"
+			// Text(SoftLineBreak): "]"
+
+			// grab these nodes
+			firstParagraph := v.FirstChild()
+			if firstParagraph.ChildCount() < 3 {
+				return ast.WalkContinue, nil
+			}
+			firstTextNode, ok := firstParagraph.FirstChild().(*ast.Text)
+			if !ok {
```

Comment:
Fail fast
```suggestion
			if !ok || string(firstTextNode.Segment.Value(reader.Source())) != "[" {
```

same for second and third node

Reply:
I did it the way it is right now for readability. Do you prefer it this way?

Reply:
Ah yes it makes sense to fail fast https://github.com/go-gitea/gitea/pull/29121/commits/03882cf0864d3a09d74b205e11f93e3fb64a90f1